src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml

```diff
@@ -30,17 +30,13 @@
 
         <!-- DependencyInjectionExtension -->
         <service id="form.extension" class="Symfony\Component\Form\Extension\DependencyInjection\DependencyInjectionExtension" public="false">
-            <argument type="service" id="service_container" />
-            <!-- All services with tag "form.type" are inserted here by FormPass -->
-            <argument type="collection" />
-            <!-- All services with tag "form.type_extension" are inserted here by FormPass -->
-            <argument type="collection" />
-            <!-- All services with tag "form.type_guesser" are inserted here by FormPass -->
-            <argument type="collection" />
+            <argument type="service-locator" /><!-- All services with tag "form.type" are stored in a service locator by FormPass -->
+            <argument type="collection" /><!-- All services with tag "form.type_extension" are stored here by FormPass -->
+            <argument type="iterator" /><!-- All services with tag "form.type_guesser" are stored here by FormPass -->
         </service>
 
         <!-- ValidatorTypeGuesser -->
-        <service id="form.type_guesser.validator" class="Symfony\Component\Form\Extension\Validator\ValidatorTypeGuesser">
+        <service id="form.type_guesser.validator" class="Symfony\Component\Form\Extension\Validator\ValidatorTypeGuesser" public="false">
```

Comment:
see PR description for why this is not a BC break (same for all added `public="false"`)